src/map/Map.cpp

```diff
@@ -0,0 +1,216 @@
+ */
+int Map::loadFileData() {
+    // Open file stream for selected map file
+    ifstream file(fname.c_str());
+
+    if (file.fail()) {
+        logv("failed to open file\n");
+        return 0;
+    }
+
+    // Array of wallstart positions
+    array<MapPoint, MAX_WALLS> wallStart;
+
+    // Indices
+    int i = 0;
+    int j = 0;
+
+    // Reading in the filestream into a buffer.
+    file.seekg(0, std::ios::end);
+    size_t size = file.tellg();
+    std::string buffer(size, ' ');
+    file.seekg(0);
+    file.read(&buffer[0], size);
+
+    for (const char& ch : buffer) {
+        switch (ch) {
+            case ',':
+                continue;
+            case '\n':
+                j = 0;
+                ++i;
+                continue;
+            case WALL:              //Wall piece found, update AI Map
+                AIMap[i][j] = 1;
+                break;
+            case WALL_START:        //Label start of wall rectangle
+                wallStart[wallStartCount].x = j;
+                wallStart[wallStartCount].y = i;
+                ++wallStartCount;
+                AIMap[i][j] = 1;
+                break;
+            // case CONCRETE_START:    //Start of concrete area
+            case ZOMBIE_SPAWN:      //Zombie spawn Points
+                zombieSpawn[zombieSpawnCount].x = j;
+                zombieSpawn[zombieSpawnCount].y = i;
+                ++zombieSpawnCount;
+                break;
+            case SHOP_SPOT:         // Shop spot points
+                shops[shopCount].x = j;
+                shops[shopCount].y = i;
+                ++shopCount;
+                break;
+            case BASE_START:        //Start of base area
+                base.x = j;
+                base.y = i;
+                break;
+        }
+        mapdata[i][j] = ch;               //Store map file value
+        ++j;
+    }
+    genWalls(wallStart);
+    return 1;
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: void genWalls(const struct MapPoint wstart[MAX_WALLS])
+ * Parameters:
+ *      wstart - Array of wall start MapPoints.
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+void Map::genWalls(const array<MapPoint, MAX_WALLS> wallStart) {
+    int nextx;
+    int nexty;
+    int endx;
+    int endy;
+
+    for (int q = 0; q < wallStartCount; q++) {
+        int startx = wallStart[q].x;
+        int starty = wallStart[q].y;
+        nextx = startx + 1;
+        nexty = starty + 1;
+        while (mapdata[starty][nextx] == WALL) {
+            nextx ++;
+        }
+        endx = nextx;
+        while (mapdata[nexty][startx] == WALL) {
+            nexty++;
+        }
+        endy = nexty;
+        MapWall temp;
+        temp.x = startx * T_SIZE;
+        temp.y = starty * T_SIZE;
+        temp.width = (endx - startx) * T_SIZE;
+        temp.height = (endy - starty) * T_SIZE;
+        walls.push_back(temp);
```

Comment:
I don't think you need to use a temp variable here
You have two choices:
Either use emplace_back, or construct a temporary with uniform initialization and push_back that
I would prefer you using emplace, but I'm not sure if a default constructor for structs exists where you can initialize all the values.
Since I don't know if you want to add such a constructor, I have also suggested the second option which uses insert.
Either way, please change this